generators/openapi-client/prompts.js

```diff
@@ -191,7 +191,7 @@ function askExistingAvailableDocs() {
     this.prompt(prompts).then(props => {
         if (props.availableDoc !== undefined) {
             this.props.inputSpec = props.availableDoc.url;
-            this.props.cliName = props.availableDoc.name;
+            this.props.cliName = props.availableDoc.name === 'default' ? 'apidocs' : props.availableDoc.name; // "default" cannot be used as it's a keyword in java
```

Comment:
if there are other forbidden keywords we could validate the cliName when "What is the unique name for your API client ?" is prompted to the user. WDYT?

Reply:
@ecostanzi : That question is shown conditionally based on `this.props.action === 'new' && !response.useServiceDiscovery`. So if we add it there, it might not be triggered for all flows of the generator. Does that make sense or were you implying something else? :smile: 

Also I was wondering (maybe I am not understanding the logic here correctly), is it strictly necessary to give the user the ability to choose the api client name? Sanitizing against all java keywords will be an interesting problem (maybe there's some library to do it); but wouldn't it be simpler to maybe just take any cliName and append it with say something like, `cliName + "_apidoc"`. This way it is guaranteed there's no unintentional keyword problems.

Reply:
> Does that make sense or were you implying something else? 
It doesn't, I'll check it

> Sanitizing against all java keywords will be an interesting problem
You're right, maybe avoiding to suggest java keywords is enough.

The client name is the unique identifier of the single generated client. It is necessary to fill the .yo-rc.json, to choose the package name etc. Also, when generating from static files or remote URIs the client name cannot be inferred unless by parsing the content of the yml file, so we let the user to choose it.

Reply:
> It doesn't, I'll check it

Sorry, maybe I should have been more clearer. What I meant was suppose when prompted "Do you want to use Eureka service discovery ?" the user chooses "true". Then the prompt "What is the unique name for your API client ?" will not appear. And thus what I mean is doing a check there for cliName is probably not suitable since the question might or might not show up. Does that makes things more clearer? :smile: 

> You're right, maybe avoiding to suggest java keywords is enough.

I've added a warning to the "What is the unique name for your API client ?" prompt make things clearer. Let me know what you think. :smile:

Reply:
> I've added a warning to the "What is the unique name for your API client ?" prompt make things clearer. Let me know what you think.

Ok, it's clearer now

> What I meant was suppose when prompted "Do you want to use Eureka service discovery ?" the user chooses "true". Then the prompt "What is the unique name for your API client ?" will not appear.

No worries your point was clear, my answer not so much. I created the submodule from this repository https://github.com/cbornet/generator-jhipster-swagger-cli/  . This configuration was [already there](https://github.com/cbornet/generator-jhipster-swagger-cli/blob/master/generators/app/index.js#L129) and to be honest I had never noticed that the cli name is not prompted when eureka service discovery is enabled. I'll try to understand if it makes sense. Thank you for pointing it out.

Reply:
@ecostanzi : No problem. I'll mark this resolved since the keyword issue (which is the intention of this PR) seems resolved now. But let me know if you have any other issues. 😄